views/js/pciCreator/dev/textReaderInteraction/pciCreator.js

```diff
@@ -23,6 +26,22 @@ define([
          * @returns {Object} Widget
          */
         getWidget : function () {
+            Widget.beforeStateInit(function (event, pci, state) {
+                if (pci.typeIdentifier && pci.typeIdentifier === "textReaderInteraction") {
+                    if (!pci.widgetRenderer) {
+                        pci.widgetRenderer = new Renderer({
+                            serial : pci.serial,
+                            $container : state.widget.$container,
+                            templates : {
+                                pages : pagesTpl,
+                                navigation : navigationTpl
+                            }
+                        });
+                    }
+                    pci.widgetRenderer.setState(state.name);
+                    pci.widgetRenderer.renderAll(pci.properties);
+                }
+            });
```

Comment:
I moved setting up `beforeStateInit` callback to the `pciCreator.js` because setting up of this callback should happen every time after creating the widget.
If this would be happen in the [/textReaderInteraction/creator/widget/Widget.js](https://github.com/oat-sa/extension-tao-itemqti-pci/blob/develop/views/js/pciCreator/dev/textReaderInteraction/creator/widget/Widget.js) (like it was before) then this callback will be setted up once after the `Widget.js` file will be loaded. 
This event setted up here:
https://github.com/oat-sa/extension-tao-itemqti/blob/master/views/js/qtiCreator/widgets/Widget.js#L179
As you can see there simply added the event to the `document` ('beforeStateInit.qti-widget....')
But if you leave authoring section (press `manage items` button) this event will be removed here:
https://github.com/oat-sa/extension-tao-itemqti/blob/master/views/js/controller/creator/main.js#L210
So, if you return to the authoring then `Widget.js` won't be loaded again and `beforeStateInit` callback won't be setted up.